### DISCUSSION TITLE
[site] Amazon URL shortening help

### DISCUSSION THREAD

**Discussion by Mark-Decken** [einaregilsson/Redirector#404] [General]
Please can someone help me.

I'm trying to shorten all Amazon URLs so they go from this:

https://www.amazon.co.uk/Electronic-Arts-1068618-Battlefield-2042/dp/B096YRJX3J?crid=1NTYYL6LZGGRR&dib=eyJ2IjoiMSJ9.Y2ur9zHAiFzDcP30ftyMnjapLwBBZblst64daWg99qrmOgDudiQYdxkehUUPXl6NWPgL2GlaRpuZbP_xqExi54k6GnDDu3Hcnjx6CsTauSIBa0F-gQ3nY-g_tO8ncfTwiNDMWId-8Pv6del1yxM90hrzVipUHLJrvpG2A0wDUJBB7TMuqT2Rugt9zAuX2aFw8glP1d49v49Njkc472EIYvGbQ78oat0BthJXo43RgIQ.4gFd9e3z73HggBSJ6K-dt8q1pq27Zxk2ftdTNb-SDuM&dib_tag=se&keywords=Battlefield+2042&nsdOptOutParam=true&qid=1730373760&sprefix=battlefield+2042%2Caps%2C156&sr=8-3

to 

https://www.amazon.co.uk/Electronic-Arts-1068618-Battlefield-2042/dp/B096YRJX3J

Thank you.

**Comment by jadedgnome** [comment]
**Include pattern:**
`^(https://www.amazon.co.uk/.*/dp/.*)\?.*
`
**Redirect to:**
$1


or

**Include pattern:**
`(^https?://(?:www\.)?amazon.[^/].*)\?.*
`
**Redirect to:**
$1

**Pattern type:**
Regular Expression

**check**
History State

**Reply by Mark-Decken** [reply]
Thank you very much.

What does history state do? I can't find any explanation for it in the help section.

Also for anyone wanting to use this redirect, add this or search won't work:

Exclude pattern: https://www.amazon.co.uk/s
